Y0/y0_euler.py

```diff
@@ -0,0 +1,294 @@
+from functools import partial
+from mpi4py import MPI
+
+from meshmode.array_context import PyOpenCLArrayContext
+from meshmode.dof_array import thaw
+from meshmode.mesh import BTAG_ALL, BTAG_NONE  # noqa
+from grudge.eager import EagerDGDiscretization
+from grudge.shortcuts import make_visualizer
+
+
+from mirgecom.euler import inviscid_operator
+from mirgecom.simutil import (
+    inviscid_sim_timestep,
+    sim_checkpoint
+)
+from mirgecom.io import (
+    make_init_message,
+)
+# from mirgecom.checkstate import compare_states
+from mirgecom.integrators import rk4_step
+from mirgecom.steppers import advance_state
+from mirgecom.boundary import (
+    PrescribedBoundary,
+    AdiabaticSlipBoundary,
+    DummyBoundary
+)
+from mirgecom.initializers import (
+    Lump,
+    make_pulse
+)
+from mirgecom.eos import IdealSingleGas
+
+
+def get_pseudo_y0_mesh():
+    """Generate or import a grid using `gmsh`.
+
+    Input required:
+        data/pseudoY0.brep  (for mesh gen)
+        -or-
+        data/pseudoY0.msh   (read existing mesh)
+
+    This routine will generate a new grid if it does
+    not find the grid file (data/pseudoY0.msh), but
+    note that if the grid is generated in millimeters,
+    then the solution initialization and BCs need to be
+    adjusted or the grid needs to be scaled up to meters
+    before being used with the current main driver in this
+    example.
+    """
+    from meshmode.mesh.io import read_gmsh, generate_gmsh, ScriptWithFilesSource
+    import os
+    if os.path.exists("data/pseudoY0.msh") is False:
+        mesh = generate_gmsh(
+            ScriptWithFilesSource("""
+            Merge "data/pseudoY0.brep";
+            Mesh.CharacteristicLengthMin = 1;
+            Mesh.CharacteristicLengthMax = 10;
+            Mesh.ElementOrder = 2;
+            Mesh.CharacteristicLengthExtendFromBoundary = 0;
+
+            // Inside and end surfaces of nozzle/scramjet
+            Field[1] = Distance;
+            Field[1].NNodesByEdge = 100;
+            Field[1].FacesList = {5,7,8,9,10};
+            Field[2] = Threshold;
+            Field[2].IField = 1;
+            Field[2].LcMin = 1;
+            Field[2].LcMax = 10;
+            Field[2].DistMin = 0;
+            Field[2].DistMax = 20;
+
+            // Edges separating surfaces with boundary layer
+            // refinement from those without
+            // (Seems to give a smoother transition)
+            Field[3] = Distance;
+            Field[3].NNodesByEdge = 100;
+            Field[3].EdgesList = {5,10,14,16};
+            Field[4] = Threshold;
+            Field[4].IField = 3;
+            Field[4].LcMin = 1;
+            Field[4].LcMax = 10;
+            Field[4].DistMin = 0;
+            Field[4].DistMax = 20;
+
+            // Min of the two sections above
+            Field[5] = Min;
+            Field[5].FieldsList = {2,4};
+
+            Background Field = 5;
+        """, ["data/pseudoY0.brep"]), 3, target_unit='MM')
```

Comment:
These files are not here. It's easy to add download code. No need to make this harder to use than necessary.